src/freak.jl

```diff
@@ -0,0 +1,105 @@
+type FREAK{S, T, O} <: DescriptorParams
+    pattern_scale::Float64
+    pattern_table::Array{Array{S, 1}, 1}
+    smoothing_table::Array{T, 1}
+    orientation_weights::Array{O, 1}
+end
+
+typealias OrientationPair Vector{Int}
+typealias SamplePair Vector{Float64}
+
+function FREAK(; pattern_scale::Float64 = 22.0)
+    pattern_table, smoothing_table = _freak_tables(pattern_scale)
+    orientation_weights = OrientationPair[]
+    for o in freak_orientation_sampling_pattern
+        offset_1 = pattern_table[1][o[1]]
+        offset_2 = pattern_table[1][o[2]]
+        dy, dx = offset_1 - offset_2
+        norm = (dx ^ 2 + dy ^ 2)
+        push!(orientation_weights, OrientationPair([round(Int, dy * 4096 / norm), round(Int, dx * 4096 / norm)]))
+    end
+    FREAK(pattern_scale, pattern_table, smoothing_table, orientation_weights)
+end
+
+function _freak_mean_intensity{T<:Gray}(int_img::AbstractArray{T, 2}, keypoint::Keypoint, offset::SamplePair, sigma::Float64)
+    y = keypoint[1] + offset[1]
+    x = keypoint[2] + offset[2]
+    if sigma < 0.5
+        tl = bilinear_interpolation(int_img, y - sigma, x - sigma)
+        tr = bilinear_interpolation(int_img, y - sigma, x + sigma)
+        bl = bilinear_interpolation(int_img, y + sigma, x)
+        br = bilinear_interpolation(int_img, y + sigma, x + sigma)
+        return (br + tl - tr - bl) / (4 * (sigma ^ 2))
```

Comment:
@timholy This wasnt covered by the test cases since default ```pattern_scale``` is 22 which scales up the ```sigma```. After checking coveralls I attempted to cover this condition too. Turns out this is not correct as bilinear interpolation of the integral image may not give correct values to calculate a ```diff```. What would be a good alternative for this? Should I just take a average of the 8x8 neighbourhood?